src/lightning/fabric/utilities/distributed.py

```diff
@@ -147,11 +150,13 @@ def _sync_ddp(result: Tensor, group: Optional[Any] = None, reduce_op: Optional[U
     # Sync all processes before reduction
     torch.distributed.barrier(group=group)
     torch.distributed.all_reduce(result, op=op, group=group, async_op=False)
+    world_size = torch.distributed.get_world_size(group)
 
-    if divide_by_world_size:
-        result = result / torch.distributed.get_world_size(group)
-
-    return result
+    if not divide_by_world_size:
+        return result
+    if not torch.is_floating_point(result):
```

Comment:
I think this is worth a comment

```suggestion
    # torch.distributed.all_reduce is inplace, so we should do the division inplace to leave the modified tensors with the expected value
    if not torch.is_floating_point(result):
```